src/Web/WsCommand/WsCmdGetDisp.cpp

```diff
@@ -77,38 +77,100 @@ void WsCmdGetDisp::execute(AsyncWebSocket* server, uint32_t clientId)
     }
     else
     {
-        IDisplay&   display     = Display::getInstance();
-        size_t      fbLength    = display.getWidth() * display.getHeight();
-        uint32_t*   framebuffer = new(std::nothrow) uint32_t[fbLength];
+        constexpr size_t fbLength    = CONFIG_LED_MATRIX_WIDTH * CONFIG_LED_MATRIX_HEIGHT;
+        uint32_t*        framebuffer = new (std::nothrow) uint32_t[fbLength];
 
         if (nullptr == framebuffer)
         {
             sendNegativeResponse(server, clientId, "\"Internal error.\"");
         }
         else
         {
-            uint32_t    index       = 0U;
-            String      msg;
-            uint8_t     slotId      = SlotList::SLOT_ID_INVALID;
+            String        msg;
+            uint8_t       slotId     = SlotList::SLOT_ID_INVALID;
+
+            uint32_t      lastColor  = 0U;    /* The color that started a repeat sequence. */
+            uint32_t      color      = 0U;    /* Actual color in read order.               */
+            size_t        index      = 1U;    /* Next value from framebuffer to used.      */
+            uint8_t       repeat     = 0U;    /* Repeat count for current color.           */
+            const uint8_t REPEAT_MAX = 0xFFU; /* Maximum repeat color counter value.       */
 
             DisplayMgr::getInstance().getFBCopy(framebuffer, fbLength, &slotId);
 
             preparePositiveResponse(msg);
-
             msg += slotId;
-            msg += DELIMITER;
-            msg += display.getWidth();
-            msg += DELIMITER;
-            msg += display.getHeight();
 
-            for(index = 0U; index <  fbLength; ++index)
+            /* RGB data is send in a "compressed" format using a repeat counter in
+             * the upper 8 bits. The send values are <rep>:<r>:<g>:<b>.
+             * The repeat counter indicates how often the same color shall be used
+             * in subsequent pixels. Use a small state machine to calculate the
+             * repeat counter.
+             *
+             * Example:
+             * A black only 32x8 framebuffer would be send as a single 0xFF000000 value.
+             *
+             */
+            enum
+            {
+                STATE_GETDISP_COLLECT = 0, /* Collect sequences of same color. */
+                STATE_GETDISP_SEND,        /* Send current color sequence.     */
+                STATE_GETDISP_FINISH       /* Terminate state machine          */
+            } state   = STATE_GETDISP_COLLECT;
+
+            lastColor = framebuffer[0];
+
+            while (state != STATE_GETDISP_FINISH)
             {
-                msg += DELIMITER;
-                msg += Util::uint32ToHex(framebuffer[index]);
+                if (STATE_GETDISP_COLLECT == state)
+                {
+                    if (index < fbLength)
+                    {
+                        color = framebuffer[index];
+                        if (color != lastColor)

```

Comment:
In the very first run and if the color is red, its different than the lastColor, because lastColor was initialized with black.

Reply:
LastColor is loaded with framebuffer[0] not black(0). See line 120. 

It would have been obvious if lastColor got also defined at line 120. But style is to define locals at block start and initialized, So it got initialized with 0 and right before the state machine updated with first real color. I could not do this at block start, because framebuffer isn't allocated there yet. 
 
 To prove, I did your use case  and ploted a red pixel at (0,0). It works as expeceted:
 
![grafik](https://github.com/user-attachments/assets/2db40d9e-d47c-41a3-86fa-3d8acc7bb8b2)

Reply:
You can skip the initialization, but keep variable definition at the begin of the block scope. Its done on several similar placess too. But I think the static check rule seems to be disabled, which would detect it.

Reply:
Ok